sdk/compute/Azure.ResourceManager.Compute/tests/Mock/MockingPatternValidationTests.cs

```diff
@@ -0,0 +1,70 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Linq;
+using System.Reflection;
+using System.Runtime.CompilerServices;
+using NUnit.Framework;
+
+namespace Azure.ResourceManager.Compute.Tests.Mock
+{
+    public class MockingPatternValidationTests
+    {
+        private const string RPNamespace = "Azure.ResourceManager.Compute";
+        private const string RPName = "Compute";
+
+        // TODO -- after enough test cases on each individual RP, we should move this to mgmt test base class so that every RP will automatically get this test case
+        [Test]
+        public void ValidateMockingPattern()
```

Comment:
This is the test case that validates the "mocking pattern" in our generated and customized source code.

It is validating in the following rule:
If there is an extension method,
1. there must be a type in namespace `{RPNamespace}.Mocking` with the name of `{RPName}{ExtendeeName}MockingExtension`.
2. the type in step 1 must have a public virtual method with the same name and the exact same parameters (exclude the first one)